packages/jsapi-utils/src/TableUtils.test.ts

```diff
@@ -267,6 +306,381 @@ describe('quick filter tests', () => {
+        '"c"'
+      );
+      testComponentFilter('null', FilterType.isNull, 'char');
+    });
+
+    it('should return a text filter for any other column type', () => {
+      testComponentFilter(
+        '\\*foo',
+        FilterType.eqIgnoreCase,
+        'notatype',
+        '*foo'
+      );
+      testComponentFilter('foo', FilterType.eqIgnoreCase, 'string', 'foo');
+    });
+  });
+
+  describe('makeAdvancedValueFilter', () => {
+    const filterTypesWithArgument: FilterTypeValue[] = [
+      FilterType.eq,
+      FilterType.eqIgnoreCase,
+      FilterType.notEq,
+      FilterType.notEqIgnoreCase,
+      FilterType.greaterThan,
+      FilterType.greaterThanOrEqualTo,
+      FilterType.lessThan,
+      FilterType.lessThanOrEqualTo,
+    ];
+
+    const filterTypesWithNoArguments: FilterTypeValue[] = [
+      FilterType.isTrue,
+      FilterType.isFalse,
+      FilterType.isNull,
+    ];
+
+    const invalidFilterTypes: FilterTypeValue[] = [
+      FilterType.in,
+      FilterType.inIgnoreCase,
+      FilterType.notIn,
+      FilterType.notInIgnoreCase,
+      FilterType.invoke,
+    ];
+
+    function testInvokeFilter(type, operation, value, timezone, ...args) {
+      const column = makeFilterColumn(type);
+      const filter = column.filter() as MockFilter;
+
+      const nullResult = makeFilterCondition();
+      const notResult = makeFilterCondition();
+      const invokeResult = makeFilterCondition();
+      const expectedResult = makeFilterCondition();
+
+      filter.isNull.mockReturnValueOnce(nullResult);
+
+      (nullResult.not as jest.Mock).mockReturnValueOnce(notResult);
+
+      filter.invoke.mockReturnValueOnce(invokeResult);
+
+      (notResult.and as jest.Mock).mockReturnValueOnce(expectedResult);
+
+      const result = TableUtils.makeAdvancedValueFilter(
+        column,
+        operation,
+        value,
+        timezone
+      );
+
+      expect(filter.isNull).toHaveBeenCalled();
+      expect(nullResult.not).toHaveBeenCalled();
+      expect(notResult.and).toHaveBeenCalledWith(invokeResult);
+
+      if (args.length > 0) {
+        expect(filter.invoke).toHaveBeenCalledWith(...args);
+      } else {
+        expect(filter.invoke).toHaveBeenCalled();
+      }
+      expect(result).toBe(expectedResult);
+    }
+
+    const testFilterWithOperation = (
+      functionName,
+      value,
+      expectedFn,
+      operation: FilterTypeValue,
+      timezone: string,
+      type: string,
+      ...args
+    ) => {
+      const column = makeFilterColumn(type);
+      const filter = column.filter();
+
+      const expectedResult = makeFilterCondition();
+
+      filter[expectedFn].mockReturnValueOnce(expectedResult);
+
+      const result = TableUtils[functionName](
+        column,
+        operation,
+        value,
+        timezone
+      );
+
+      if (args.length > 0) {
+        expect(filter[expectedFn]).toHaveBeenCalledWith(...args);
+      } else {
+        expect(filter[expectedFn]).toHaveBeenCalled();
+      }
```

Comment:
Don't split it out, let's be explicit about what the expected function gets called with.